doc/adr/0004-project-versioning.md

```diff
@@ -0,0 +1,29 @@
+# Project versioning with SNAPSHOT's
+
+Date: 2022-01-25
+
+## Status
+
+Proposed
+
+## Context
+
+We need a versioning scheme for the project that will allow simultaneous development of the library modules (`kernel` and `sdk`) and the projects in the `examples/` directory that can depend 
+on the library modules. SNAPSHOT versioning will allow this.
+
+## Decision
+
+1. `main` branch contains changes that have been released
+2. `develop` branch contains changes that haven't been released yet
+3. Project version on the `main` branch follow semver's scheme `MAJOR.MINOR.PATCH`  
```

Comment:
I'd suggest:
```md
3. Project versioning follows [semver](https://semver.org)
4. `master` branch must contain version without precedence (`a.b.c-alpha` or `a.b.c-SNAPSHOT` is not allowed)
5. `develop` branch can contain version with precedence

### Release flow

1. Create branch from the `develop` or feature branch: `release/vX.Y.Z`
2. Create commit with updated version for the release matching the release branch name
(example: branch: `release/v0.1.0` with version update from `0.1.0-SNAPSHOT` to `0.1.0`)
3. Merge release branch to `master` using `git merge --ff-only`.
4. Merge `master` branch to `develop` using `git merge --ff-only`
5. Create commit with an updated version on `develop` branch that increments MINOR part with SNAPSHOT precedence
(example: after `0.1.0` release update to `0.2.0-SNAPSHOT`)

### Hotfixes

1. Create branch from the given tag or `master` branch: `hotfix/something-is-broken`
2. Apply fix
2. Create commit with updated version for the hotfix release by increasing PATCH part
(example: version update from `0.1.2` to `0.1.3`)
3. Merge hotfix branch to `master` using `git merge --ff-only`.
4. Merge `master` branch to `develop` using `git merge --ff-only`
5. Resolve conflicts and ignore hotfix version update

```

Reply:
I think that instructions for the release or for the hotfixing **should not** be part of the ADR so I intentionally left it out.

Reply:
Why? It's something we need to agree on. Especially it can be a consequence of this ADR

Reply:
It's not a consequence of this ADR. This ADR is about versioning with SNAPSHOT, that's all. The branching strategy is independent from the versioning scheme used.